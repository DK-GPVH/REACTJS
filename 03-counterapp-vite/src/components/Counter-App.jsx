import { useState } from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({value})=>{
    const [ Counter,setCounter] = useState(value);

    const add =()=>{
        // console.log("hola");
        //setCounter(Counter + 1);
        setCounter(Counter+1);
    };
    const sub = ()=>{
        setCounter(Counter-1);
    };
    const reset = ()=>{
        setCounter(value);
    }
    return <>
        <h1>CounterAPP</h1>
        <h2>{ Counter}</h2>

        <button onClick={add}> ADD +</button>
        <button onClick={sub}> SUBTRACT -</button>
        <button aria-label='btn-reset' onClick={reset}> RESET Â°</button>

    </>
}

CounterApp.propTypes = {
    value : PropTypes.number.isRequired
}

CounterApp.defaultProps = {
    value : "No enviaste ningun valor"
}
