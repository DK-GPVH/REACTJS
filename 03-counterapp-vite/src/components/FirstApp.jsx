import PropTypes from "prop-types";

const objeto = {
    nombre : "Jorge",
    email : "@gmail.com",
    edad : 45
}

function funcion () {
    return console.log("hola");
}
export const FirstApp=(props)=>{
    //<Fragment> => <>
    return (
        //no se puede mandar objetos {}
        //para enviar objetos mediante stringify
        <>
        {/* <h1>{"Hola"}FirstApp</h1> */}
        {/* <p>{props.title}</p>
        <h1>{props.title}</h1>
        <h2>{JSON.stringify(objeto)}</h2>
        <h3>{funcion()}</h3>         */}
        {/*crando un data-testid*/}
        <h1 data-testid="test-title">{props.title}</h1>
        <p>{props.subtitle}</p>
        <p>{props.nombre}</p>
        </> 
        

    )
};

//definir los props por defecto
FirstApp.defaultProps = {
    subtitle:'No hay subtitulo',
    nombre:'no hay nombre',
    // title : 'No hay titulo',
}

//Atributos de los props
FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string
}

