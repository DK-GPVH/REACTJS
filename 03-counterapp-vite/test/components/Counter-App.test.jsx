import { fireEvent, render, screen } from "@testing-library/react";
import { CounterApp } from "../../src/components/Counter-App";

describe('Pruebas en counter app',()=>{
    const value=100;

    test('Validar la similitud con screenshot ',()=>{
        const {container} = render(<CounterApp value={value}/>);
        expect(container).toMatchSnapshot();
    });


    test('Validar el mostrar el valor inicial',()=>{
        render(<CounterApp value={value}/>);
        expect(screen.getByText('100')).toBeTruthy();
    });

    test('Debe de incrementar al hacer click en el botton +',()=>{
        render(<CounterApp value={value}></CounterApp>);
        fireEvent.click(screen.getByText('ADD +'))
        expect(screen.getByText('101')).toBeTruthy();
    });

    test('Debe de decrementar al hacer click en el botton -',()=>{
        render(<CounterApp value={value}></CounterApp>);
        fireEvent.click(screen.getByText('SUBTRACT -'))
        expect(screen.getByText('99')).toBeTruthy();
    });

    test('Debe de funcionar reset',()=>{
        render( <CounterApp value={value}/>);
        fireEvent.click(screen.getByText('ADD +'));
        fireEvent.click(screen.getByText('ADD +'));
        fireEvent.click(screen.getByText('ADD +'));
        //llamar por rol
        fireEvent.click(screen.getByRole('button',{name:'btn-reset'}));
        // fireEvent.click(screen.getByText('RESET Â°'));
        expect(screen.getByText(100)).toBeTruthy();
        // expect(screen.getByText('100')).toBeTruthy();
    });
})