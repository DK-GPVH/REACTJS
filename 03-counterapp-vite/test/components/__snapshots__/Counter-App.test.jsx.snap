// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en counter app Validar la similitud con screenshot  1`] = `
<div>
  <h1>
    CounterAPP
  </h1>
  <h2>
    100
  </h2>
  <button>
     ADD +
  </button>
  <button>
     SUBTRACT -
  </button>
  <button
    aria-label="btn-reset"
  >
     RESET Â°
  </button>
</div>
`;
